### ISSUE TITLE
Too long name on PVC volume

### ISSUE THREAD

**Issue by haf-afa** [coreos/prometheus-operator#2865]
This one was closed ahead of time: https://github.com/coreos/prometheus-operator/issues/2853 without investigating why or how.

I tried to deploy with a smaller name:

```
# Source: prometheus-operator/templates/prometheus/prometheus.yaml
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: prometheus
  labels:
    app: prometheus
    component: prometheus

```

But the same error happens:

```
monitoring   0s          Warning   Failed                  pod/prometheus-prometheus-0                                              Error: stat /host_mnt/c/Users/hefe01/.docker/Volumes/prometheus-prometheus-db-prometheus-prometheus-0/pvc-e9c6b1c1-0a02-11ea-8e2d-00155dfa1319: no such file or directory

```

**Comment by brancz** [comment]
I do think this is a problem, but how do we prevent breaking existing users here, should we introduce a new naming scheme?

**Comment by haf-afa** [comment]
It may be specific to how docker-desktop works, since it happens on macOS ^^^ here. macOS probably needs the hostpath provisioner.

**Comment by brancz** [comment]
Let me rephrase, I think the current naming scheme is unnecessarily long, meaning we should try to somehow make it either as minimal as possible or make it completely configurable.

**Comment by rmartinez3** [comment]
Not sure if read this issue correctly . But you are allowed to configure the actual pvc for prometheus and alertmanager via the volumeclaim template 
```
volumeClaimTemplate:
      metadata:
        name: data
```
hence actually making the pvc volume name short 

I had this issue when i first was deploying prometheus operator to use pvc and gluster. Since there was a character limit. That way that should fix your problem. However, when upgrading  statefulset have to delete statefulset first since statefulset does not allow you to upgrade and change pvc fields.

**Comment by haf-afa** [comment]
@brancz I think smaller name by default is a great idea.

@rmartinez3 Thank you for the tip, let me try that. My current hypothesis is that since I tried both on macOS' and Windows' docker-desktop and had the problem on both, the issue lies elsewhere than length limitations, so trying will let me rule that in.

**Comment by haf-afa** [comment]
.............

**Comment by RubyLangdon** [comment]
I've got the same problem and setting the name does not seem to be working. 
This is in my values file:
```
  prometheus:
    prometheusSpec:
      storageSpec:
        volumeClaimTemplate:
          metadata:
            name: the-name-i-want
```
And this is what I get in the prometheus resource:
```
  storage:
    volumeClaimTemplate:
      metadata: {}
```
The metadata is not getting set so the default hugely long name is being used.

**Comment by Neraud** [comment]
@RubyLangdon I've had the same problem and I've opened a dedicated issue : #3093

**Comment by surajssd** [comment]
Should this be closed now that the issue https://github.com/coreos/prometheus-operator/issues/3093 is fixed? And there is a new release as well https://github.com/coreos/prometheus-operator/releases/tag/v0.39.0
